### PR TITLE
fix: fix tests failed on windows

### LINKED ISSUE #3150: Release blocker of v0.6
### What type of enhancement is this?

Other

### What does the enhancement do?

This issue tracks the release progress of v0.6.

- [x] #3146 
- [x] #3148 
- [x] #3149
- [x] #3154
- [ ] #3155

### Implementation challenges

_No response_

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

fix tests failed on windows.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

## Refer to a related PR or issue link (optional)
close #3150

### CODE REVIEW

**Comment by WenyXu** [comment]
Wait for CI on the Windows https://github.com/GreptimeTeam/greptimedb/actions/runs/7498917198

**Comment by killme2008** [comment]
Looks like there are several places have to be reverted

https://github.com/GreptimeTeam/greptimedb/commit/8bd4a36136b#diff-cccb6171cdcd0330fef0f232d470cfee6d72cb3f6830e72e8e4c9e27d0b728c1

File: src/object-store/src/util.rs [hunk 1]
```diff
@@ -44,8 +34,24 @@ pub fn normalize_dir(dir: &str) -> String {
 pub fn join_dir(parent: &str, child: &str) -> String {
     // Always adds a `/` to the output path.
     let output = format!("{parent}/{child}/");
-    // We call opendal's normalize_root which keep the last `/`.
-    opendal::raw::normalize_root(&output)
+    normalize_dir(&output)
+}
+
+/// Modified from the `opendal::raw::normalize_root`
```
Comment (evenyag) [inline src/object-store/src/util.rs:40]: Also document the behavior.

File: src/object-store/src/util.rs [hunk 2]
```diff
@@ -44,8 +34,24 @@ pub fn normalize_dir(dir: &str) -> String {
...
+    let has_root = v.starts_with('/');
+    let mut v = v
+        .split('/')
+        .filter(|v| !v.is_empty())
+        .collect::<Vec<&str>>()
+        .join("/");
+    if has_root {
+        v.insert(0, '/');
+    }
+    if !v.ends_with('/') {
```
Comment (v0y4g3r) [inline src/object-store/src/util.rs:51]: We can skip this check since `v` never ends with `/`.
```suggestion
    v.push('/')
```

**Review by MichaelScofield — APPROVED** [review]

File: src/object-store/src/util.rs [hunk 3]
```diff
@@ -44,8 +34,42 @@ pub fn normalize_dir(dir: &str) -> String {
...
+/// # The different
+///
+/// It doesn't always append `/` ahead of the path,
+/// It only keeps `/` ahead if the original path starts with `/`.
+///
+/// Make sure root is normalized to style like `/abc/def/`.
+///
+/// # Normalize Rules
+///
+/// - All whitespace will be trimmed: ` abc/def ` => `abc/def`
```
Comment (v0y4g3r) [inline src/object-store/src/util.rs:51]: Here the doc contradict with the actual behavior, ` abc/def ` will be normalized to `abc/def/` as per the implementation.

File: src/object-store/src/util.rs [hunk 4]
```diff
@@ -44,8 +34,42 @@ pub fn normalize_dir(dir: &str) -> String {
...
+}
+
+/// Modified from the `opendal::raw::normalize_root`
+///
+/// # The different
+///
+/// It doesn't always append `/` ahead of the path,
+/// It only keeps `/` ahead if the original path starts with `/`.
+///
+/// Make sure root is normalized to style like `/abc/def/`.
```
Comment (evenyag) [inline src/object-store/src/util.rs:47]:
```suggestion
/// Make sure the directory is normalized to style like `abc/def/`.
```

**Review by evenyag — APPROVED** [review]
Rest LGTM

File: src/object-store/src/util.rs (hunk 3 above)
Comment (WenyXu) [reply src/object-store/src/util.rs:51]: It is just one of the rules; the function will apply all rules and yield the final result.

**Review by v0y4g3r — APPROVED** [review]
LGTM

**Pull request merged**